astropy/nddata/utils.py

```diff
@@ -916,3 +925,373 @@ def bbox_cutout(self):
         are for the valid (non-filled) cutout values.
         """
         return self._calc_bbox(self.slices_cutout)
+
+
+def make_cutouts(data, catalog, wcs=None, origin=0, verbose=True):
+    """Make cutouts of catalog targets from a 2D image array.
+
+    Notes
```

Comment:
The `Notes` section should go below the `Returns` section (we follow the numpydoc docstring style).